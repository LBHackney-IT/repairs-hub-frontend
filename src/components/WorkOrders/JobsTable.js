import PropTypes from 'prop-types'
import JobRow from './JobRow'

const JobsTable = ({ workOrders }) => (
  <div>
    <p className="govuk-heading-s">Manage jobs</p>

    <table className="govuk-table govuk-!-margin-top-5 govuk-!-width-full hackney-work-order-table">
      <thead className="govuk-table__head">
        <tr className="govuk-table__row govuk-body">
          <th scope="col" className="govuk-table__header">
            Reference
          </th>
          <th scope="col" className="govuk-table__header">
            Date raised
          </th>
          <th scope="col" className="govuk-table__header">
            Last update
          </th>
          <th scope="col" className="govuk-table__header">
            Priority
          </th>
          <th scope="col" className="govuk-table__header">
            Property
          </th>
          <th scope="col" className="govuk-table__header">
            Status
          </th>
          <th scope="col" className="govuk-table__header">
            Description
          </th>
          <th scope="col" className="govuk-table__header"></th>
        </tr>
      </thead>
      <tbody className="govuk-table__body">
        {workOrders.map((job, index) => (
          <JobRow key={index} {...job} />
        ))}
      </tbody>
    </table>
  </div>
)

JobsTable.propTypes = {
  workOrders: PropTypes.arrayOf(
    PropTypes.shape({
      reference: PropTypes.string.isRequired,
      dateRaised: PropTypes.instanceOf(Date),
      lastUpdated: PropTypes.instanceOf(Date),
      priority: PropTypes.string,
      property: PropTypes.string.isRequired,
      status: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
    })
  ).isRequired,
}

export default JobsTable
