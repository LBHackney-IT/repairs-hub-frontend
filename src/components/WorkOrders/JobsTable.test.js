import { render } from '@testing-library/react'
import JobsTable from './JobsTable'

describe('JobsTable component', () => {
  const props = [
    {
      reference: '00012345',
      dateRaised: new Date('2021-01-12T16:24:26.632Z'),
      lastUpdated: new Date('2021-01-13T16:24:26.632Z'),
      priority: 'E - Emergency (24 hours)',
      property: '1 Pitcairn House St Thomass Square',
      owner: '',
      description:
        'ALPHA- Pitcairn house op stucl behind carpark gates from power network pls remedy AND Communal: Door entry; Residents locked out/in',
    },
  ]

  it('should render properly', () => {
    const { asFragment } = render(<JobsTable workOrders={props} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
