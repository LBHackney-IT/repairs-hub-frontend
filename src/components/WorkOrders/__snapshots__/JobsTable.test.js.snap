// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobsTable component should render properly 1`] = `
<DocumentFragment>
  <div>
    <p
      class="govuk-heading-s"
    >
      Manage jobs
    </p>
    <table
      class="govuk-table govuk-!-margin-top-5 govuk-!-width-full hackney-work-order-table"
    >
      <thead
        class="govuk-table__head"
      >
        <tr
          class="govuk-table__row govuk-body"
        >
          <th
            class="govuk-table__header"
            scope="col"
          >
            Reference
          </th>
          <th
            class="govuk-table__header"
            scope="col"
          >
            Date raised
          </th>
          <th
            class="govuk-table__header"
            scope="col"
          >
            Last update
          </th>
          <th
            class="govuk-table__header"
            scope="col"
          >
            Priority
          </th>
          <th
            class="govuk-table__header"
            scope="col"
          >
            Property
          </th>
          <th
            class="govuk-table__header"
            scope="col"
          >
            Description
          </th>
        </tr>
      </thead>
      <tbody
        class="govuk-table__body"
      >
        <tr
          class="govuk-table__row govuk-table__row--clickable govuk-body-s"
        >
          <td
            class="govuk-table__cell"
          >
            00012345
          </td>
          <td
            class="govuk-table__cell"
          >
            12 Jan 2021
            <div
              class="work-order-hours"
            >
              4:24 pm
            </div>
          </td>
          <td
            class="govuk-table__cell"
          >
            13 Jan 2021
            <div
              class="work-order-hours"
            >
              4:24 pm
            </div>
          </td>
          <td
            class="govuk-table__cell"
          >
            E - Emergency (24 hours)
          </td>
          <td
            class="govuk-table__cell"
          >
            1 Pitcairn House St Thomass Square
          </td>
          <td
            class="govuk-table__cell"
          >
            <p
              class="description"
            >
              ALPHA- Pitcairn house op stucl behind carpark gates from power network pls remedy AND Communal: Door entry; Residents locked out/in
            </p>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
